snowflake/exec.go

```diff
@@ -1,28 +1,73 @@
 package snowflake
```

Comment:
This file has the main changes. Each function here prepares the SQL statement unless the # of args is 0.

Reply:
I also added ctx as the first argument to all of these functions